imap_processing/glows/l1/glows_l1a_data.py

```diff
@@ -0,0 +1,411 @@
+        self.de_data = bytearray(level0.DE_DATA)
+        self.missing_seq = []
+
+        self.block_header = {
+            "ground_software_version": version,
+            "pkts_file_name": self.l0.packet_file_name,
+            # note: packet number is seq_count (per apid!) field in CCSDS header
+            "seq_count_in_pkts_file": self.l0.ccsds_header.SRC_SEQ_CTR,
+        }
+
+        if level0.LEN == 1:
+            self._process_de_data()
+
+    def append(self, second_l0: DirectEventL0):
+        """Merge an additional direct event packet to this DirectEventL1A class.
+
+        Direct event data can span multiple packets, as marked by the SEQ and LEN
+        attributes. This method will add the next piece of data in the sequence
+        to this data class. The two packets are compared with
+        DirectEventL0.sequence_match_check. If they don't match, the method throws
+        a ValueError.
+
+        If the sequence is broken, the missing sequence numbers are added to
+        missing_seq. Once the last value in the sequence is reached, the data is
+        processed from raw bytes to useful information.
+
+        Parameters
+        ----------
+        second_l0: DirectEventL0
+            Additional L0 packet to add to the DirectEventL1A class
+        """
+        # if SEQ is missing or if the sequence is out of order, do not continue.
+        if not second_l0.SEQ or second_l0.SEQ < self.most_recent_seq:
+            raise ValueError(
+                f"Sequence for direct event L1A is out of order or "
+                f"incorrect. Attempted to append sequence counter "
+                f"{second_l0.SEQ} after {self.most_recent_seq}."
+            )
+
+        # Track any missing sequence counts
+        if second_l0.SEQ != self.most_recent_seq + 1:
+            self.missing_seq.extend(range(self.most_recent_seq + 1, second_l0.SEQ))
+
+        # Determine if new L0 packet matches existing L0 packet
+        match = self.l0.sequence_match_check(second_l0)
+
+        # TODO: Should this raise an error? Log? something else?
+        if not match:
+            raise ValueError(
+                f"While attempting to merge L0 packet {second_l0} "
+                f"into L1A packet {self.__repr__()}, mismatched values "
+                f"were found. "
+            )
+
+        self.de_data.extend(bytearray(second_l0.DE_DATA))
+
+        self.most_recent_seq = second_l0.SEQ
+        # if this is the last packet in the sequence, process the DE data
+        # TODO: What if the last packet never arrives?
+        if self.l0.LEN == self.most_recent_seq + 1:
+            self._process_de_data()
+
+    def _process_de_data(self):
+        """
+        Process direct event bytes.
+
+        Once the packets are complete, create the status data table from the first 40
+        bytes in de_data, and the direct events from the remaining bytes.
+        """
+        self.status_data = self._generate_status_data(self.de_data[:40])
```

Comment:
Another possible idea here would be to turn `self.status_data` into a property and move the content of these functions up there into the property function definition. Fine as you have it, so just an idea to consider if you like.